8932 container base image (#8933)
IQSS/dataverse: doc/sphinx-guides/source/container/index.rst

Container Guide
===============

**Contents:**

.. toctree::

>   base-image

Reviewer: Sorry, sorry, I was thinking this is a page, not a whole new guide. All set.

> Running Dataverse software in containers is quite different than in a :doc:`classic installation <../installation/prep>`.

Author: :heavy_check_mark:

> Both approaches have pros and cons. These days (2022) containers are very often used for development and testing,

Reviewer 2: Remove (2022), unless you're sure that you'll remember to edit it in later years (more hassle). I agree that containers are becoming ever more present and I don't see that changing very soon.

Author: ```suggestion
Both approaches have pros and cons. These days containers are very often used for development and testing,
```

> Both approaches have pros and cons. These days (2022) containers are very often used for development and testing,

Reviewer 2: ```suggestion
Both approaches have pros and cons. These days, containers are very often used for development and testing,
```

> but there is an ever rising move for running applications in the cloud using container technology.

Author: :heavy_check_mark:

**NOTE:**
**As the Institute for Quantitative Social Sciences (IQSS) at Harvard is running their installations in the classic
deployment way, the container support is mostly created and maintained by the Dataverse community on a best-effort
basis.**

> This guide is *not* about installation on technology like Docker Swarm, Kubernetes, Rancher or other

Reviewer 3: ```suggestion
This guide does **not** describe how to run nor build the application with Docker. It only describes how to create a base image for creating application images (executable versions) and how to configure each one. Executable container images are described for example here (`Dataverse on K8s project <https://k8s-docs.gdcc.io>`_).   
```

Reviewer 3: Somehow I can not add a multiline suggestion. The rest of the paragraph can also be removed.

Author: You need to select multiple lines first by clicking on the line number where you want to start, hold shift and click on the last line number.

I'll go ahead and resolve this comment and recreate it as multiline.

> This guide is *not* about installation on technology like Docker Swarm, Kubernetes, Rancher or other
> solutions to run containers in production. There is the `Dataverse on K8s project <https://k8s-docs.gdcc.io>`_ for this
> purpose.

Author: By @johannes-darms 

```suggestion
This guide does **not** describe how to run nor build the application with Docker. It only describes how to create a base image for creating application images (executable versions) and how to configure each one. Executable container images are described for example here (`Dataverse on K8s project <https://k8s-docs.gdcc.io>`_).   
```

Author: Re-reading this: What you suggest adding here might be in the wrong place. This is the main container guide page, not the base image page. Please re-route. Thanks!

This guide focuses on describing the container images managed from the main Dataverse repository (again: by the
community, not IQSS), their features and limitations. Instructions on how to build the images yourself, how to

> extend them and how to use them for development purposes may be found in respective subpages.

Author: Yes, we agreed on a new guide.

I'll rephrase a bit to underline this is about "developing containers" rather than "developing Dataverse".
